packages/core/src/agent/message-list/index.ts

```diff
@@ -1068,16 +1060,22 @@ export class MessageList {
       if (latestMessage.createdAt.getTime() < messageV2.createdAt.getTime()) {
         latestMessage.createdAt = messageV2.createdAt;
       }
-      if (!latestMessage.content.content && messageV2.content.content) {
-        latestMessage.content.content = messageV2.content.content;
-      }
-      if (
-        latestMessage.content.content &&
-        messageV2.content.content &&
-        latestMessage.content.content !== messageV2.content.content
-      ) {
-        // Match what AI SDK does - content string is always the latest text part.
-        latestMessage.content.content = messageV2.content.content;
+      // Extract text content from parts for comparison and update
+      const latestTextContent = latestMessage.content.parts.find(p => p.type === 'text')?.text;
+      const messageV2TextContent = messageV2.content.parts.find(p => p.type === 'text')?.text;
+
+      if (!latestTextContent && messageV2TextContent) {
+        // Add text part if it doesn't exist
+        const textPartIndex = latestMessage.content.parts.findIndex(p => p.type === 'text');
+        if (textPartIndex === -1) {
+          latestMessage.content.parts.unshift({ type: 'text', text: messageV2TextContent });
+        }
+      } else if (latestTextContent && messageV2TextContent && latestTextContent !== messageV2TextContent) {
+        // Update existing text part with new content
+        const textPart = latestMessage.content.parts.find(p => p.type === 'text');
+        if (textPart && textPart.type === 'text') {
+          textPart.text = messageV2TextContent;
+        }
```

Comment:
I don't think this makes sense, we shouldn't be updating text parts, the whole point is you can have multiple text parts in the array and they're ordered in the order they happened in. This code looks like it assumes there's only a single text part. Maybe we can just delete all of this code that was added? All new parts should be added into the end of the parts array. The only time we should mutate parts is when we get a tool result and need to update a tool invocation to change the state and add the result